subprojects/platform-jvm/src/test/groovy/org/gradle/api/java/archives/internal/DefaultAttributesTest.groovy

```diff
@@ -46,22 +46,32 @@ class DefaultAttributesTest extends Specification {
         attributes.putAll((null): 'someValue')
 
         then:
-        thrown(ManifestException)
+        thrown(NullPointerException)
     }
 
     def testValueValidationWithPut() {
         when:
         attributes.put('key', null)
 
         then:
-        thrown(ManifestException)
+        thrown(NullPointerException)
     }
 
     def testValueValidationWithPutAll() {
         when:
         attributes.putAll('key': null)
 
         then:
-        thrown(ManifestException)
+        thrown(NullPointerException)
+    }
+
+    def caseSensitivity() {
+        when:
+        attributes.put("Hello", "world")
+        attributes.put("hello", "keys are case-insensitive")
+
+        then:
+        attributes.get("Hello") == "keys are case-insensitive"
+        attributes.get("hello") == "keys are case-insensitive"
```

Comment:
❌ This seems like a potential breakage that we may need to account for. Either accept it if it's fundamentally broken or nag about the change of behaviour in the next major release.

Reply:
@lacasseio , `java.util.jar.Attributes` javadoc says:

>Valid attribute names are case-insensitive

Gradle's `DefaultAttributes` pretend to `implement java.util.jar.Attributes`, so it should adhere to the interface contract.

Reply:
Oh. I missed that `DefaultAttributes` implements `org.gradle.api.java.archives.Attributes` which is not really connected with `java.util.jar.Attributes`.

However, the manifest specification says that `Attribute names are case insensitive`, so Gradle should probably adhere to that somehow. Looking forward to hearing from you on your decision if Gradle wants to support case-insensitive attribute names.